### PR TITLE
fix: add atomic version in the analytics initial config for Headless

### PR DESCRIPTION
**Pull request by louis-bompart**
Also, append instead of prepend Headless version to replace Headless-Atomic-Relay order by Atomic-Headless-Relay

https://coveord.atlassian.net/browse/KIT-3308

### CODE REVIEW

File: packages/atomic/src/components/common/interface/analytics-config.ts [hunk 1]
```diff
@@ -55,17 +55,18 @@ export function getNextAnalyticsConfig(
...
-      ...searchEngineConfig.analytics,
-    };
-  }
-  return defaultConfiguration;
+  const analyticsConfiguration = searchEngineConfig.analytics ?? {};
+  Object.assign(
+    analyticsConfiguration,
+    augmentAnalyticsConfigWithAtomicVersion()
+  );
+  Object.assign(configuration, searchEngineConfig.analytics);
```
Comment (alexprudhomme) [inline packages/atomic/src/components/common/interface/analytics-config.ts:69]: why object.assign and no spread operator ?

File: packages/atomic/src/components/common/interface/analytics-config.ts [hunk 2]
```diff
@@ -55,17 +55,18 @@ export function getNextAnalyticsConfig(
...
   };
 
-  if (searchEngineConfig.analytics) {
-    return {
-      ...defaultConfiguration,
-      ...searchEngineConfig.analytics,
-    };
-  }
-  return defaultConfiguration;
+  const analyticsConfiguration = searchEngineConfig.analytics ?? {};
```
Comment (alexprudhomme) [inline packages/atomic/src/components/common/interface/analytics-config.ts:64]: Also isnt analyticsConfiguration useless here ? It is assigned and then unused

**Review by Spuffynism — APPROVED** [review]

File: packages/atomic/src/components/common/interface/analytics-config.ts (hunk 1 above)
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/common/interface/analytics-config.ts:69]:
```suggestion
  Object.assign(configuration, analyticsConfiguration);
```

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/atomic/src/components/common/interface/analytics-config.ts (hunk 1 above)
Comment (louis-bompart) [reply packages/atomic/src/components/common/interface/analytics-config.ts:69]: Taste mostly. To use the spread operator here, I would need to have something like:
```
{
  ...defaultConfiguration,
  analytics: {
     ...analyticsConfiguration,
     ...augmentAnalyticsConfigWithAtomicVersion()
  }
}
```
I think it's less readable than the Object.assign

File: packages/atomic/src/components/common/interface/analytics-config.ts (hunk 2 above)
Comment (louis-bompart) [reply packages/atomic/src/components/common/interface/analytics-config.ts:64]: resolved w/ fred suggestion

**Pull request merged**